route.go

```diff
@@ -88,6 +89,40 @@ func (r *Route) GlobalMiddleware(middlewares ...httpcontract.Middleware) {
 	r.setMiddlewares(middlewares)
 }
 
+func (r *Route) Listen(l net.Listener) error {
+	r.outputRoutes()
+	color.Green().Println(termlink.Link("[HTTP] Listening and serving HTTP on", "http://"+l.Addr().String()))
+
+	r.server = &http.Server{
+		Addr:           l.Addr().String(),
+		Handler:        http.AllowQuerySemicolons(r.instance),
+		MaxHeaderBytes: r.config.GetInt("http.drivers.gin.header_limit", 4096) << 10,
+	}
+
+	if err := r.server.Serve(l); !errors.Is(err, http.ErrServerClosed) {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Route) ListenTLS(l net.Listener, certFile, keyFile string) error {
```

Comment:
Add `ListenTLS` and `ListenTLSWithCert` method? consistent with `RunTLS` and `RunTLSWithCert`?